wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -0,0 +1,268 @@
+  * confirmation counts and marking UTXOs as spent when
+  * spending from our wallet
+  */
+private[wallet] trait TransactionProcessing { self: LockedWallet =>
+  /////////////////////
+  // Public facing API
+
+  /** @inheritdoc */
+  override def processTransaction(
+      transaction: Transaction,
+      confirmations: Int): Future[LockedWallet] = {
+    logger.info(
+      s"Processing transaction=${transaction.txIdBE} with confirmations=$confirmations")
+    processTransactionImpl(transaction, confirmations).map {
+      case ProcessTxResult(outgoing, incoming) =>
+        logger.info(
+          s"Finished processing of transaction=${transaction.txIdBE}. Relevant incomingTXOs=${incoming.length}, outgoingTXOs=${outgoing.length}")
+        this
+    }
+
+  }
+
+  private[wallet] case class ProcessTxResult(
+      updatedIncoming: List[SpendingInfoDb],
+      updatedOutgoing: List[SpendingInfoDb])
+
+  /////////////////////
+  // Internal wallet API
+
+  /**
+    * Processes TXs originating from our wallet.
+    * This is called right after we've signed a TX,
+    * updating our UTXO state.
+    */
+  private[wallet] def processOurTransaction(
+      transaction: Transaction,
+      confirmations: Int): Future[ProcessTxResult] = {
+    logger.info(
+      s"Processing TX from our wallet, transaction=${transaction.txIdBE} with confirmations=$confirmations")
+    processTransactionImpl(transaction, confirmations).map { result =>
+      val txid = transaction.txIdBE
+      val changeOutputs = result.updatedIncoming.length
+      val spentOutputs = result.updatedOutgoing.length
+
+      logger.info(
+        s"Processing of internal transaction=$txid resulted in changeOutputs=$changeOutputs and spentUTXOs=$spentOutputs")
+      result
+    }
+  }
+
+  /////////////////////
+  // Private methods
+
+  /** Does the grunt work of processing a TX.
+    * This is called by either the internal or public TX
+    * processing method, which logs and transforms the
+    * output fittingly.
+    */
+  private def processTransactionImpl(
+      transaction: Transaction,
+      confirmations: Int): Future[ProcessTxResult] = {
+
+    val incomingTxoFut: Future[Vector[SpendingInfoDb]] =
+      spendingInfoDAO
+        .findTx(transaction)
+        .flatMap {
+          // no existing elements found
+          case Vector() =>
+            processNewIncomingTx(transaction, confirmations).map(_.toVector)
+
+          case txos: Vector[SpendingInfoDb] =>
+            val txoProcessingFutures =
+              txos
+                .map(processExistingIncomingTxo(transaction, confirmations, _))
+
+            Future
+              .sequence(txoProcessingFutures)
+
+        }
+
+    val outgoingTxFut: Future[Vector[SpendingInfoDb]] = {
+      for {
+        outputsBeingSpent <- spendingInfoDAO.findOutputsBeingSpent(transaction)
+        processed <- FutureUtil.sequentially(outputsBeingSpent)(
+          markAsSpentIfUnspent)
+      } yield processed.flatten.toVector
```

Comment:
So what does the `None` case mean that is being returned from `markAsSpentIfUnspent`? Does that mean that utxo was already spent? This seems like a log worth event, as it means that someone is trying to double spend a transaction that we funded, right? 

Perhaps the log need not happen here, but I think we should definitely log in `markAsSpentIfUnspent`

Reply:
It only means that the TXO was already spent. Added a couple of logs at `debug` and `trace` level to `markAsSpentIfUnspent`